src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AppointmentService.kt

```diff
@@ -290,4 +307,19 @@ class AppointmentService(
     }
     return appointmentDeliveryAddress
   }
+
+  private fun setAttendanceAndBehaviourIfHistoricAppointment(
+    appointment: Appointment,
+    appointmentTime: OffsetDateTime,
+    attended: Attended?,
+    additionalAttendanceInformation: String?,
+    behaviourDescription: String?,
+    notifyProbationPractitioner: Boolean?,
+    updatedBy: AuthUser
+  ) {
+    if (appointmentTime.isBefore(OffsetDateTime.now())) {
```

Comment:
I don't see the need in having this check here - simply set the attendance/behaviour fields based on if they're provided or not.

Reply:
Certain fields are mandatory when historic and not applicable when a future appointment. Subsequent lines utilise !! which would fail for future appointments

Reply:
I guess there are two things here: validation and functionality.

You can have functionality available to all requests based on attendance/behaviour provided but validation based on it being an historic appointment.

It just seems that it would be a bit unexpected for the API to specific that you can provide attendance/behaviour details as part of the POST request but it will be ignored if it's non-historic.

Reply:
The AppointmentValidator would kick out an inconsistent request wrt attendance and behaviour fields